### PR TITLE
Refactor iceberg class pull request monitor

### PR DESCRIPTION
**Pull request by imtayadeway**
* Pulls the worker logic out of private methods and into public methods on the repo model and stateless services so that it can be more easily tested. 
* Adds specs for all the above
* Adds factory girl to aid in model tests

### CODE REVIEW

File: app/models/commit_monitor_repo.rb [hunk 1]
```diff
@@ -64,4 +64,24 @@ def enabled_for?(checker)
...
     fq_name.in?(repos)
   end
+
+  def pr_branches
+    branches.select(&:pull_request?)
+  end
+
+  def current_pr_branch_names
+    with_git_service do |git|
+      break pull_requests.collect { |pr| git.pr_branch(pr.number) }
```
Comment (Fryguy) [inline app/models/commit_monitor_repo.rb]: break is not needed.

File: app/models/commit_monitor_repo.rb [hunk 2]
```diff
@@ -64,4 +64,24 @@ def enabled_for?(checker)
...
+    end
+  end
+
+  def pull_requests
+    with_github_service do |github|
+      break github.pull_requests.all
```
Comment (Fryguy) [inline app/models/commit_monitor_repo.rb]: Same

File: spec/workers/pull_request_monitor/pr_branch_record_spec.rb [hunk 3]
```diff
@@ -0,0 +1,106 @@
+require "spec_helper"
+
+RSpec.describe PullRequestMonitor::PrBranchRecord do
+  describe ".create" do
+    it "has git create a local pr branch" do
+      repo = spy("CommitMonitorRepo")
```
Comment (Fryguy) [inline spec/workers/pull_request_monitor/pr_branch_record_spec.rb]: Why spy vs double?  I never used spy before, and it seems inconsistent to introduce a new thing unless it has some different purpose.

**Comment by Fryguy** [comment]
Looks nice!  One rubocop warning there, btw.

File: spec/workers/pull_request_monitor/pr_branch_record_spec.rb (hunk 3 above)
Comment (imtayadeway) [inline spec/workers/pull_request_monitor/pr_branch_record_spec.rb]: it has a distinct purpose which i find quite useful in tests like these where i'm trying to keep the setup phase under control. it's basically syntactic sugar for `double("CommitMonitorRepo").as_null_object`. i tend to use this quite a lot when creating doubles for collaborators whose use are just noisy implementation details which are not relevant to the particular test at hand. any thoughts?

**Comment by imtayadeway** [comment]
@Fryguy thanks! i chose not to correct that warning as i found it helpful to give that object a 'name' by assigning it to an unused variable to better reveal its purpose.

File: app/models/commit_monitor_repo.rb (hunk 1 above)
Comment (imtayadeway) [inline app/models/commit_monitor_repo.rb]: nice! did not realize that would work

File: spec/workers/pull_request_monitor/pr_branch_record_spec.rb (hunk 3 above)
Comment (Fryguy) [inline spec/workers/pull_request_monitor/pr_branch_record_spec.rb]: My only thought was that spies are really for after-the-fact expectation checks, so it seems weird to use them as_null_object, but it is clever :smile:  I'm fine with it for now.

**Pull request merged**